### ISSUE TITLE
Endless Range Need

### ISSUE THREAD

**Issue by Madao-3** [artichoke/artichoke#441]
One of the new feature of 2.6.3 is endless range, I can do this too.
```
2.6.3 :001 > a = (20...)
 => 20...
2.6.3 :002 > a.size
 => Infinity
```

**Comment by lopopolo** [comment]
This may require extending the parser in C to support this.

**Comment by Madao-3** [comment]
actually the parer know it is a range, but mruby is not support endless range.

**Comment by lopopolo** [comment]
mruby added beginless and endless ranges in https://github.com/mruby/mruby/issues/5085.

Revendoring mruby will close this ticket, but revendoring is blocked on unforking `mrb_array` functions.

**Comment by lopopolo** [comment]
This requires revendoring to mruby 3.0.0+, specifically a ref containing mruby/mruby@bec4d053400c3a11c8efd68c3e8bd5ea4a0bcc54.

**Comment by lopopolo** [comment]
See #957 for a discussion on upgrading to mruby-3.0.0.

**Comment by lopopolo** [comment]
This will be fixed by #1125.
